src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -105,10 +116,22 @@ public void OpenPin(int pinNumber, PinMode mode)
         /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
         public void ClosePin(int pinNumber)
         {
-            int logicalPinNumber = GetLogicalPinNumber(pinNumber);
+            ClosePin(pinNumber, NumberingScheme);
+        }
+
+        /// <summary>
+        /// Closes an open pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number.</param>
+        /// <param name="numberingScheme">Numbering scheme for the given pin</param>
+        /// <remarks>Internal use, to make Dispose work consistently</remarks>
+        protected virtual void ClosePin(int pinNumber, PinNumberingScheme numberingScheme)
```

Comment:
isn't numbering scheme global per GpioController? (maybe I missed this change somewhere in here)

Reply:
Yes, it currently is. The reason for this extra overload is the Dispose method, because we only know the logical numbers of the open pins. Also, this would probably be needed if adding the IGpioPin interface, as callback on dispose. 

Maybe it should even be public, then.

Reply:
@pgrawehr I think if GpioController was created with specific numberingScheme then you should use that all the way, you should not close pins which are not managed by the gpio controller

Reply:
Already discussed and solved separtely